src/java/org/apache/cassandra/index/sai/README.md

```diff
@@ -0,0 +1,115 @@
+<!---
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+ 
+     http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-->
+
+# Storage-Attached Indexing
+
+## Overview
+Storage-attached indexing is a column based secondary indexing apparatus for Cassandra.
+
+The project was inspired by SASI (SSTable-Attached Secondary Indexes) and retains some of its high-level
+architectural character (and even some actual code), but makes significant improvements in a number of areas:
+
+- The on-disk/SSTable index formats for both string and numeric data have been completely replaced. Strings are indexed
+  on disk using our proprietary on-disk byte-ordered trie data structure, while numeric types are indexed using Lucene's
+  balanced kd-tree.
+- While indexes continue to be managed at the column level from the user's perspective, the storage design at the column
+  index level is row-based, with related offset and token information stored only once at the SSTable level. This
+  drastically reduces our on-disk footprint when several columns are indexed on the same table.
+- Tracing, metrics, virtual table-based metadata and snapshot-based backup/restore are supported out of the box.
+
+Many similarities with standard secondary indexes remain:
+
+- The full set of C* consistency levels is supported for both reads and writes.
+- Index updates are synchronous with mutations and do not require any kind of read-before-write.
+- Queries are implemented on the back of C* range reads.
+- Paging is supported.
+- Only token ordering of results is supported.
+- Index builds are visible to operators as compactions and are executed on compaction threads.
+- All DML and DDL statements are CQL-based.
+- Single-node management operations are available via nodetool. (ex. stop & rebuild_index)
+
+## Quick Start
+
+The following short tutorial will get you up-and-running with storage-attached indexing.
+
+### Build and Start Cassandra
+
+Follow the instructions to build and start Cassandra in README.asc in root folder of the Cassandra repository
+
+### Create a Simple Data Model
+
+1.) Run the following DDL statements to create a table and two indexes:
+
+`CREATE KEYSPACE test WITH replication = {'class': 'NetworkTopologyStrategy' , 'Cassandra': '1'};`
```

Comment:
@maedhroz this should be probably "datacenter1" which is default. This query will likely fail when used as is in vanilla Cassandra. Maybe using SimpleStrategy instead would be even easier.

Reply:
Up to @mike-tr-adamson , but `SimpleStrategy` WFM.

Reply:
Thanks for pointing that out. A throwback to older times. I have changed it to `SimpleStrategy`.